src/CppUTest/MemoryLeakDetector.cpp

```diff
@@ -66,7 +68,7 @@ void SimpleStringBuffer::addMemoryDump(const void* memory, size_t memorySize)
 	size_t p;
 
     while (currentPos < memorySize) {
-        add("    %04lx: ", currentPos);
+        add("    %04lx: ", MAYBE_16BIT(currentPos));
```

Comment:
@basvodde  take a look at the commit history. All of the more straightforwad "solutions" were rejected by Clang. For good reasons, even. I got really frustrated, and this is the epitome of portable code :D. So please, feel free to better it, but it does have to work!

(Technically, even this contains dead code, but the compiler doesn't see it)